internal/nmc/helper.go

```diff
@@ -74,3 +74,45 @@ func (h *helper) GetModuleEntry(nmc *kmmv1beta1.NodeModulesConfig, modNamespace,
 	}
 	return nil, 0
 }
+
+func findModuleStatus(statuses []kmmv1beta1.NodeModuleStatus, moduleNamespace, moduleName string) *kmmv1beta1.NodeModuleStatus {
+	for i := 0; i < len(statuses); i++ {
+		s := statuses[i]
+
+		if s.Namespace == moduleNamespace && s.Name == moduleName {
+			return &statuses[i]
+		}
+	}
+
+	return nil
+}
+
+func RemoveModuleStatus(statuses *[]kmmv1beta1.NodeModuleStatus, modNamespace, modName string) {
+	if statuses == nil || len(*statuses) == 0 {
```

Comment:
what if findModuleStatus also returns an index? Then you can call it from the RemoveModuleStatus and implement removal by playing with index.WDYT?

Reply:
I was trying to mimick the [apimachinery code](https://github.com/kubernetes/apimachinery/blob/master/pkg/api/meta/conditions.go) which instead builds a new slice.